lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -61,18 +61,17 @@ UsdUndoRenameCommand::UsdUndoRenameCommand(const UsdSceneItem::Ptr& srcItem, con
     else
     {
         // account for internal vs external references
+        // internal references (references without a file path specified) from the same file
+        // should be renamable.
         if (prim.HasAuthoredReferences()) {
             auto primSpec = MayaUsdUtils::getPrimSpecAtEditTarget(_stage, prim);
-            for (const SdfReference& ref : primSpec->GetReferenceList().GetAddedOrExplicitItems()) {
-                // GetAssetPath returns the asset path to the root layer of the referenced layer
-                // this will be empty in the case of an internal reference.
-                if (ref.GetAssetPath().empty()) { 
-                    return;
-                }else{
-                   std::string err = TfStringPrintf("Unable to rename referenced object [%s]", 
-                                                    prim.GetName().GetString().c_str());
-                   throw std::runtime_error(err.c_str());
-                }
+
+            if(MayaUsdUtils::isInternalReference(primSpec)) {
```

Comment:
Nit-pick: could just be
if (!MayaUsdUtils::isInternalReference(primSpec)) {
    throw blablabla;
}

Reply:
@ppt-adsk see 80aa35e4160b7c35147042f45c49d727ac3289f5